packages/web-client/app/components/card-pay/deposit-workflow/transaction-amount/index.ts

```diff
@@ -116,46 +119,54 @@ class CardPayDepositWorkflowTransactionAmountComponent extends Component<Workflo
     }
   }
 
-  @action unlock() {
-    let tokenSymbol = this.args.workflowSession.state.depositSourceToken;
-    this.isUnlocking = true;
-    taskFor(this.layer1Network.approve)
-      .perform(this.amountAsBigNumber, tokenSymbol)
-      .then((transactionReceipt: TransactionReceipt) => {
-        this.isUnlocked = true;
-        this.unlockTxnReceipt = transactionReceipt;
-      })
-      .finally(() => {
-        this.isUnlocking = false;
-      });
+  @action async unlock() {
+    this.errorMessage = '';
+
+    try {
+      this.isUnlocking = true;
+      let transactionReceipt = await taskFor(
+        this.layer1Network.approve
+      ).perform(this.amountAsBigNumber, this.currentTokenSymbol);
+      this.isUnlocked = true;
+      this.unlockTxnReceipt = transactionReceipt;
+    } catch (e) {
```

Comment:
Probably should also log the error, to help with debugging.

Reply:
Makes sense!